kratos/spatial_containers/specialized_spatial_search.h

```diff
@@ -0,0 +1,807 @@
+    // BinsDynamicObjects
+};
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class PointObject
+ * @ingroup KratosCore
+ * @brief Custom Point container to be used by the search
+ * @details It stores the pointer of a certain object
+ * @author Vicente Mataix Ferrandiz
+ */
+template<class TObject>
+class PointObject
+    : public Point
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Base class definition
+    typedef Point BaseType;
+
+    /// Counted pointer of PointObject
+    KRATOS_CLASS_POINTER_DEFINITION( PointObject );
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructors
+    PointObject():
+        BaseType()
+    {
+    }
+
+    /**
+     * @brief Constructor with object
+     * @param pObject The pointer to the object
+     */
+    PointObject(typename TObject::Pointer pObject):
+        mpObject(pObject)
+    {
+        UpdatePoint();
+    }
+
+    ///Copy constructor  (not really required)
+    PointObject(const PointObject& rRHS):
+        BaseType(rRHS),
+        mpObject(rRHS.mpObject)
+    {
+    }
+
+    /// Destructor.
+    ~PointObject() override= default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This function updates the database, using as base for the coordinates the condition center
+     */
+    void UpdatePoint();
+
+    ///@}
+    ///@name Acess
+    ///@{
+
+    /**
+     * @brief Returns the geometry associated to the point
+     * @return mrGeometry The reference to the geometry associated to the point
+     */
+    typename TObject::Pointer pGetObject()
+    {
+        return mpObject;
+    }
+
+    /**
+     * @brief Sets the object associated to the point
+     * @param pObject The pointer to the object
+     */
+    void pSetObject(typename TObject::Pointer pObject)
+    {
+        mpObject = pObject;
+        UpdatePoint();
+    }
+
+    ///@}
+private:
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    typename TObject::Pointer mpObject = nullptr;
+
+    ///@}
+
+}; // Class PointObject
+
+/**
+* @class SpecializedSpatialSearch
+* @ingroup KratosCore
+* @brief This class is used to search for elements, conditions and nodes in a given model part
+* @details In order to perform the search it uses as backend some of the the spatial containers defined `spatial_containers` folder
+* @tparam TSearchBackend The spatial container to be used as backend
+* @author Vicente Mataix Ferrandiz
+*/
+template<SpatialContainer TSearchBackend>
+class KRATOS_API(KRATOS_CORE) SpecializedSpatialSearch
+    : public SpatialSearch
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of SpecializedSpatialSearch
+    KRATOS_CLASS_POINTER_DEFINITION(SpecializedSpatialSearch);
+
+    /// The base type
+    using BaseType = SpatialSearch;
+
+    /// Common Defines
+    using PointType = BaseType::PointType;
+
+    using ElementsContainerType = BaseType::ElementsContainerType;
```

Comment:
```suggestion
    using BaseType::ElementsContainerType;
```

valid for all other typedefs that redeclare types from the base class as well

Reply:
Okay